sms.go

```diff
@@ -0,0 +1,172 @@
+package gosms
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/binary"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// DefaultSMSBytes is the default SMS size in bytes
+	DefaultSMSBytes             int = 140
+	shortReferenceInfoElementID int = 0x00
+	longReferenceInfoElementID  int = 0x08
+	udhByteLengthShort          int = 6
+	udhByteLengthLong           int = 7
+	byteLength                  int = 8
+)
+
+// SMS structure with correctly sized message and appropriate UDH
+type SMS struct {
+	from    string
+	to      string
+	content string
+	encoder string
+	udh     string
+}
+
+// newSMS initializes a new SMS
+func newSMS(from string, to string, content string, encoder string, udh string) SMS {
+	return SMS{
+		from:    from,
+		to:      to,
+		content: content,
+		encoder: encoder,
+		udh:     udh,
+	}
+}
+
+// GetFrom returns the SMS's from field
+func (s *SMS) GetFrom() string {
+	return s.from
+}
+
+// GetTo returns the SMS's to field
+func (s *SMS) GetTo() string {
+	return s.to
+}
+
+// GetContent returns the SMS's content field
+func (s *SMS) GetContent() string {
+	return s.content
+}
+
+// GetEncoder returns the SMS's encoder field
+func (s *SMS) GetEncoder() string {
+	return s.encoder
+}
+
+// GetUDH returns the SMS's udh field
+func (s *SMS) GetUDH() string {
+	return s.udh
+}
+
```

Comment:
Can we create a type, say ConcatenatedSMS, that can be initialized with an Encoder and a reference to short/long and/or a maximum message length? That'll let us have a single function, Split, that creates our array of SMS types; if we don't have one of the above set we'll just use the defaults. It just helps us have a cleaner API where we have a single method to do the splitting using the configured settings, vs. having multiple like we do now. Maybe we can sync offline and discuss how we plan on passing in the encoder :)